impeller/renderer/backend/vulkan/command_pool_vk_unittests.cc

```diff
@@ -0,0 +1,107 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/testing/testing.h"  // IWYU pragma: keep.
+#include "fml/synchronization/waitable_event.h"
+#include "impeller/renderer/backend/vulkan/resource_manager_vk.h"
+#include "impeller/renderer/backend/vulkan/test/mock_vulkan.h"
+
+namespace impeller {
+namespace testing {
+
+TEST(CommandPoolRecyclerVKTest, GetsACommandPoolPerThread) {
+  auto const context = CreateMockVulkanContext();
+
+  // Record the memory location of each pointer to a command pool.
+  int* pool1 = nullptr;
+  int* pool2 = nullptr;
+
+  // Create a command pool in two threads and record the memory location.
+  std::thread thread1([&]() {
+    auto const pool = context->GetCommandPoolRecycler()->Get();
+    pool1 = reinterpret_cast<int*>(pool.get());
+  });
+
+  std::thread thread2([&]() {
+    auto const pool = context->GetCommandPoolRecycler()->Get();
+    pool2 = reinterpret_cast<int*>(pool.get());
+  });
+
+  thread1.join();
+  thread2.join();
+
+  // The two command pools should be different.
+  EXPECT_NE(pool1, pool2);
+}
+
+TEST(CommandPoolRecyclerVKTest, GetsTheSameCommandPoolOnSameThread) {
+  auto const context = CreateMockVulkanContext();
+
+  auto const pool1 = context->GetCommandPoolRecycler()->Get();
+  auto const pool2 = context->GetCommandPoolRecycler()->Get();
+
+  // The two command pools should be the same.
+  EXPECT_EQ(pool1.get(), pool2.get());
+}
+
+namespace {
+
+// Invokes the provided callback when the destructor is called.
+//
+// Can be moved, but not copied.
```

Comment:
Can you patch `fml::ScopedAutoResetClosure` to be move safe instead?

Reply:
I could, but I want to limit how much we're changing in this PR.

Filed https://github.com/flutter/flutter/issues/134568.